pkg/frontend/frontend.go

```diff
@@ -60,3 +71,14 @@ func NewServer(jsonRPC server.JSONRPC) *Server {
 		},
 	}
 }
+
+// NewServerWithSubsystemListener creates initialized instance of FrontEnd server communicating
+// with provided jsonRPC and externally created SubsystemListener instead default one.
+func NewServerWithSubsystemListener(jsonRPC server.JSONRPC, sysListener SubsystemListener) *Server {
+	if sysListener == nil {
+		log.Panic("nil for SubsystemListener is not allowed")
+	}
```

Comment:
What do you think about panic vs return err here ?

Reply:
This method is used only at wiring stage when we start our server and we should clearly indicate that something is going wrong and the system will be inconsistent. Error can be ignored, but panic requires reactions.